File: <100/013/013.py
def main() -> None:
    filename = '013.txt'
    a: int = 0
    with open(filename, 'r') as f:
        for row in f:
            if row[-1] == '\n':
                row = row[:-1]
            a += int(row)

    answer = str(a)[:10]
    print(answer)

    return


if __name__ == '__main__':
    main()

File: <100/013/013.hs
import System.IO ( hClose, hGetContents, openFile, IOMode(ReadMode) )


main :: IO ()
main = cat filepath

cat:: FilePath -> IO()
cat filename = do
    handle     <- openFile filename ReadMode
    contents   <- hGetContents handle
    putStrLn . take numDigits . show . sum . map readInteger . lines $ contents 
    hClose handle


----------------------- supplemental -----------------------
filepath :: String
filepath = "013.txt"


readInteger :: String -> Integer
readInteger = read


numDigits :: Int
numDigits   = 10
